txscript/engine.go

```diff
@@ -1060,6 +1107,40 @@ func (vm *Engine) Execute() (err error) {
 
 			return dstr + astr
 		}))
+
+		if vm.StepCallback != nil {
+			scriptIdx := vm.scriptIdx
+			opcodeIdx := vm.opcodeIdx
+
+			// In case the execution has completed, we keep the
+			// current script index while increasing the opcode
+			// index. This is to indicate that no new script is
+			// being executed.
+			if done {
```

Comment:
Not entirely sure what you mean? Move this whole callback step AFTER `CheckErrorCondition` has been called?

Reply:
I think what he means is that we could avoid the `done` branch in this loop if you have the callback inside `CheckErrorCondition` because you know the Engine is done. 

imo how it's implemented here is fine, I think it's better to just have every change in this one function (if what I said above is what roasbeef meant).

Reply:
Calling `ScriptCallback` from within `CheckErrorCondition` means an additional place where we have to repeat this code. I think doing it like this is simpler tbh.